starter-code/src/functions-and-arrays.js

```diff
@@ -1,16 +1,69 @@
 // Iteration #1: Find the maximum
 
+const maxOfTwoNumbers = (num1, num2) => {
+  return Math.max(num1, num2);
+}
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+const findLongestWord = (words) => {
+  let arrayLength = [];
+  if (words.length !== 0) {
+    words.forEach(element => {
+      arrayLength.push(element.length);
+    });
+    // console.log(arrayLength);
+
+    let longest = Math.max(...arrayLength);
+    // console.log(longest);
+
+    // will find the first occurence of the word that has the same length
+    let indexLongest = arrayLength.indexOf(longest);
+    // console.log(indexLongest);
+
+    return words[indexLongest];
+  } else {
+    return null;
+  }
```

Comment:
Muito bom, apenas duas dicas:
1) Não tem nada de errado com seu If-Else, porém, é mais usual a gente testar a condição anormal no If do que deixar lá em baixo na função ex: **if (words.length === 0) return null** -- e depois disso vem o resto da função.

2) Em vez pegar o length de cada palavra, jogar em um array, fazer o Math.max, depois achar o index desse length e aí retornar a palavra, poderiam ter seguido por uma abordagem onde você inicializa uma variável com uma string vazia, itera o array e só pergunta se o length da palavra no array é maior do que o da variável, se for, re-atribui o valor da variável pra ser a palavra, se não for, a variável fica como está. É legal que estão exercitando várias coisas como Spread, Math.Max, indexOf etc, mas vale pensar um pouco nessa outra abordagem também. (desculpe o textão =] )